src/main/kotlin/no/nav/familie/ef/sak/vilkår/VurderingService.kt

```diff
@@ -166,6 +174,97 @@ class VurderingService(
         tilbakestillEndretTidForKopierteVurderinger(kopiAvVurderinger, tidligereVurderinger)
     }
 
+    @Transactional
+    fun gjenbrukInngangsvilkårVurderinger(
+        nåværendeBehandlingId: UUID,
+        tidligereBehandlingId: UUID
+    ) {
+        validerBehandlingForGjenbruk(
+            nåværendeBehandlingId,
+            tidligereBehandlingId
+        )
+        val forrigeBarnIdTilNåværendeBarnMap =
+            finnBarnPåBeggeBehandlinger(nåværendeBehandlingId, tidligereBehandlingId)
+        val sivilstandErLik =
+            hentGrunnlagsdataOgValiderSivilstand(nåværendeBehandlingId, tidligereBehandlingId)
+        val tidligereVurderinger = hentVurderingerSomSkalGjenbrukes(
+            sivilstandErLik,
+            tidligereBehandlingId,
+            forrigeBarnIdTilNåværendeBarnMap
+        )
+        val nåværendeVurderinger =
+            vilkårsvurderingRepository.findByBehandlingId(nåværendeBehandlingId)
+        val vurderingerSomSkalLagres = lagInngangsvilkårVurderingerForGjenbruk(
+            nåværendeBehandlingId,
+            nåværendeVurderinger,
+            tidligereVurderinger,
+            forrigeBarnIdTilNåværendeBarnMap
+        )
+        secureLogger.info(
+            "${SikkerhetContext.hentSaksbehandler()} gjenbruker vurderinger fra behandling $tidligereBehandlingId " +
+                "for å oppdatere vurderinger på inngangsvilkår for behandling $nåværendeBehandlingId"
+        )
+        vilkårsvurderingRepository.updateAll(vurderingerSomSkalLagres)
+    }
+
+    private fun lagInngangsvilkårVurderingerForGjenbruk(
+        behandlingId: UUID,
+        nåværendeVurderinger: List<Vilkårsvurdering>,
+        tidligereVurderinger: List<Vilkårsvurdering>,
+        forrigeVurderingIdTilNåværendeBarnMap: Map<UUID, BehandlingBarn>,
+    ) = tidligereVurderinger.mapNotNull { tidligereVurdering ->
+        val barnForVurdering = forrigeVurderingIdTilNåværendeBarnMap[tidligereVurdering.barnId]
+        nåværendeVurderinger.firstOrNull { it.type == tidligereVurdering.type && it.barnId == barnForVurdering?.id }
```

Comment:
Dette funker sikkert, men vi har en hjelpemetode på vilkår `gjelderFlereBarn` som kan brukes for å se om du trenger å bry deg om barn eller ikke. 

Ikke sikker på om det blir ryddigere, men nå ser det _nesten_ ut som alle må ha `barnId` for å bli kopiert

Reply:
Jeg synes ikke det ble noe mer oversiktelig ved å diffe på `gjelderFlereBarn` men ser at det ser ut som at man må `barnId` så jeg slenger inn en liten kommentar i metoden istedet :)